In [8]:
from sklearn.ensemble import RandomForestClassifier
import pandas as pd

In [3]:
df = pd.read_csv("../data/united/dataset.csv")

In [4]:
df.head()

,PassengerId,HomePlanet,CryoSleep,Cabin,Destination,Age,VIP,RoomService,FoodCourt,ShoppingMall,Spa,VRDeck,Name,Transported
0,0001_01,Europa,False,B/0/P,TRAPPIST-1e,39.0,False,0.0,0.0,0.0,0.0,0.0,Maham Ofracculy,False
1,0002_01,Earth,False,F/0/S,TRAPPIST-1e,24.0,False,109.0,9.0,25.0,549.0,44.0,Juanna Vines,True
2,0003_01,Europa,False,A/0/S,TRAPPIST-1e,58.0,True,43.0,3576.0,0.0,6715.0,49.0,Altark Susent,False
3,0003_02,Europa,False,A/0/S,TRAPPIST-1e,33.0,False,0.0,1283.0,371.0,3329.0,193.0,Solam Susent,False
4,0004_01,Earth,False,F/1/S,TRAPPIST-1e,16.0,False,303.0,70.0,151.0,565.0,2.0,Willy Santantines,True


In [5]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 12970 entries, 0 to 12969
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   PassengerId   12970 non-null  str    
 1   HomePlanet    12682 non-null  str    
 2   CryoSleep     12660 non-null  object 
 3   Cabin         12671 non-null  str    
 4   Destination   12696 non-null  str    
 5   Age           12700 non-null  float64
 6   VIP           12674 non-null  object 
 7   RoomService   12707 non-null  float64
 8   FoodCourt     12681 non-null  float64
 9   ShoppingMall  12664 non-null  float64
 10  Spa           12686 non-null  float64
 11  VRDeck        12702 non-null  float64
 12  Name          12676 non-null  str    
 13  Transported   12970 non-null  bool   
dtypes: bool(1), float64(6), object(2), str(5)
memory usage: 1.3+ MB


In [6]:
df.isnull().sum()

PassengerId       0
HomePlanet      288
CryoSleep       310
Cabin           299
Destination     274
Age             270
VIP             296
RoomService     263
FoodCourt       289
ShoppingMall    306
Spa             284
VRDeck          268
Name            294
Transported       0
dtype: int64

Dependiendo de los nulos se borrara la fila o se rellenara con lo que sea :P

- **Homeplanet**: 288 nulos que podemos poner como "NotNow"
- **CryoSleep**: 310 nulos de gente que a lo mejor se criogenizó, 
- **Destination**: 274 nulos, como no se si lo que paso fue antes de x parada lo pondre como importante
- **Age**: 270 nulos, es importante, en el original salvaban a los niños
- **VIP**: 296 nulos, puede ser importante dependiendo de los provolegios que tuviera
- **RoomService**: 263 nulos, gastos que han echo X, puede ser importante
- **FoodCourt**: 289 nulos, gastos que han echo X, puede ser importante
- **ShoppingMall**: 306 nulos, gastos que han echo X, puede ser importante
- **Spa**: 284 nulos, gastos que han echo X, puede ser importante
- **VRDeck**: 268 nulos, gastos que han echo X, puede ser importante

- **Cabin**: 299 nulos, el numero de cabina creo que da igual
- **Name**: 294 nulos que dan igual porque son el nombre

Planazo: quitar todos los nulos y ver las cols mas importantes para ver que hacer con ellas

In [ ]:
# X = variables
X = df.drop("PassengerId", axis=1)

# y = objetivo
y = df["PassengerId"]

# convertir categóricas si hace falta
X = pd.get_dummies(X)

modelo = RandomForestClassifier()
modelo.fit(X, y)

importancias = pd.DataFrame({
    'columna': X.columns,
    'importancia': modelo.feature_importances_
})

importancias = importancias.sort_values(
    by='importancia',
    ascending=False
)

print(importancias.head(10))

c:\Users\albaro\AppData\Local\Programs\Python\Python313\Lib\site-packages\sklearn\ensemble\_forest.py:830: UserWarning: The number of unique classes is greater than 50% of the number of samples. `y` could represent a regression problem, not a classification problem.
  check_classification_targets(y)
c:\Users\albaro\AppData\Local\Programs\Python\Python313\Lib\site-packages\sklearn\tree\_classes.py:297: UserWarning: The number of unique classes is greater than 50% of the number of samples. `y` could represent a regression problem, not a classification problem.
  check_classification_targets(y)
c:\Users\albaro\AppData\Local\Programs\Python\Python313\Lib\site-packages\sklearn\tree\_classes.py:297: UserWarning: The number of unique classes is greater than 50% of the number of samples. `y` could represent a regression problem, not a classification problem.
  check_classification_targets(y)
c:\Users\albaro\AppData\Local\Programs\Python\Python313\Lib\site-packages\sklearn\tree\_classes.py:297: